monosi/anomalies/__init__.py

```diff
@@ -28,12 +28,10 @@ def _std_dev(cls, values: List[float]):
 
     @classmethod
     def _z_scores(cls, stats: List[MetricStat], filter_none=True):
-        stats_to_calc = stats
+        stats = list(filter(lambda x: x == None, stats))
 
-        if filter_none:
-            stats_to_calc = [stat for stat in stats if stat.value != None]
-
-        values = [stat.value for stat in stats_to_calc]
+        stat_values: List[float | Any] = [stat.value for stat in stats]
+        values: List[float] = list(filter(lambda x: x == None, stat_values))
```

Comment:
But what about the `filter_none` parameter? Either we do a conditional statement here or remove it...maybe something like this would work:
```
values: List[float] = list(filter(lambda x: x == None, stat_values)) if filter_none else stat_values
```

Reply:
We're going to end up removing it completely with the removal of Optional float on metricstat

Your example wouldn't work because we need values to be List[float]

Reply:
sounds good 👍